scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -113,6 +113,10 @@ def add_parse_arguments(parser = None):
                              help="""List tests with duplicate identifiers.
         """)
 
+    case_select.add_argument("--test-duplicates-as-error", action="store_true",
+                             help="""Terminate with error when duplicate identifiers.
+        """)
```

Comment:
Should this default to true? IMHO, twister should have always been treating this as an error.

Reply:
I also think it should be default. However, I don't want to break CIs of other people, since those can contain duplicates in tests not in zephyr. The last commit adds this flag to zephyr's on-pr checks, to prevent new duplicates being added to tree.
The option --list-test-duplicates is available for a while. Maybe there was another reason why duplicates are not errors by default? @nashif?

Reply:
let's make it the default.